### PR TITLE
Implement CS checking based on the `WP_CLI_CS` ruleset

### LINKED ISSUE #168: Adopt and enforce new `WP_CLI_CS` standard
We have a new PHPCS standard for WP-CLI called `WPCliCS` (props @jrfnl). It is part of the `wp-cli/wp-cli-tests` package starting with version v2.1.0.

To adopt & enforce this new standard, the following actions need to be taken for this repository:

- [x] Create a PR that adds a custom ruleset `phpcs.xml.dist` to the repository

    - Add `phpcs.xml.dist` file
    - Adapt `.distignore` to ignore `phpcs.xml.dist` & `phpunit.xml.dist`
    - Adapt `.gitignore` to ignore `phpunit.xml`, `phpcs.xml` & `.phpcs.xml`
    - Require version `^2.1` of the `wp-cli/wp-cli-tests` as a dev dependency
- [ ] Make any required changes to the code that fail the checks from the above ruleset in separate PRs
- [ ] Merge thre ruleset once all required changes have been processed and merged

A sample PR for a simple repository can be seen here: https://github.com/wp-cli/maintenance-mode-command/pull/3/files

Related https://github.com/wp-cli/wp-cli/issues/5179

### PR DESCRIPTION
**Pull request by thrijith**
Add a PHPCS ruleset using the new `WP_CLI_CS` standard.

Fixes #168

Related https://github.com/wp-cli/wp-cli/issues/5179

### CODE REVIEW

File: src/WP_CLI/CommandWithUpgrade.php [hunk 1]
```diff
@@ -195,9 +204,9 @@ public function install( $args, $assoc_args ) {
 					$errors++;
 				} elseif ( is_wp_error( $result ) ) {
 					$key = $result->get_error_code();
-					if ( in_array( $key, array( 'plugins_api_failed', 'themes_api_failed' ) )
-						&& ! empty( $result->error_data[ $key ] ) && in_array( $result->error_data[ $key ], array( 'N;', 'b:0;' ) ) ) {
-						\WP_CLI::warning( "Couldn't find '$slug' in the WordPress.org {$this->item_type} directory." );
+					if ( in_array( $key, [ 'plugins_api_failed', 'themes_api_failed' ], true )
+						&& ! empty( $result->error_data[ $key ] ) && in_array( $result->error_data[ $key ], [ 'N;', 'b:0;' ], true ) ) {
```
Comment (thrijith) [inline src/WP_CLI/CommandWithUpgrade.php]: I am unsure about this check over here, `in_array( $result->error_data[ $key ], [ 'N;', 'b:0;' ], true )`, I tried to install non existent plugin, but was not able to get this result.

Comment (schlessera) [reply src/WP_CLI/CommandWithUpgrade.php]: Here's the commit in question. There's an associated test that should still pass then...

Comment (schlessera) [reply src/WP_CLI/CommandWithUpgrade.php]: Forgot the actual link: https://github.com/wp-cli/extension-command/commit/cf8631cb6d800ae4dfd32221735a93f5667406c0

Comment (thrijith) [reply src/WP_CLI/CommandWithUpgrade.php]: Ok, only the test was changed in https://github.com/wp-cli/extension-command/pull/85 to adopt a new version of API.

Comment (schlessera) [reply src/WP_CLI/CommandWithUpgrade.php]: Leave it in with the strict check for now, It didn't seem to break something up until now, and I don't think the type can be off in this case.

**Pull request merged**